core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/offer/ContractDefinitionServiceImpl.java

```diff
@@ -91,7 +94,7 @@ private boolean evaluatePolicies(ContractDefinition definition, ParticipantAgent
     private Result<Policy> evaluate(String policyId, ParticipantAgent agent) {
         return Optional.of(policyId)
                 .map(policyStore::findById)
-                .map(policy -> policyEngine.evaluate(NEGOTIATION_SCOPE, policy, agent))
+                .map(policy -> policyEngine.evaluate(NEGOTIATION_SCOPE, policy.getPolicy(), agent))
```

Comment:
`policy.getPolicy()` gets pretty weird to read, I would move this getter on a `.map` function, to have this written as `.map(PolicyDefinition::getPolicy)`